experimental/prompt-rewrite-eval/evaluator.py

```diff
@@ -0,0 +1,199 @@
+    "java": ["java"],
+    "javascript_typescript": ["js", "jsx", "mjs", "ts", "tsx", "mts"],
+    "rust": ["rs"],
+    "php": ["php", "php3", "php4", "php5", "phps", "phpt"],
+    "lua": ["lua"]
+}
+
+# Handy index class to help get completion requests from sample repository.
+class Index():
+    # Could use a skiplist to implement the Index.
+    # But since typically total numbers of files in a repository
+    # is not likely to be a lot, a simple array should suffice.
+    def __init__(self):
+        self.arr = []
+        self.dic = {}
+    
+    # Need to guarantee the keys in add operations are
+    # always monotonously incremental.
+    def add(self, key, val):
+        self.arr.append(key)
+        self.dic[key] = val
+
+    def get(self, key):
+        pk = 0
+        for k in self.arr:
+            if k > key:
+                # Return file and the line number in the file
+                return self.dic[k], key - pk
+            pk = k
+        return self.dic[k], key - pk
+
+def wait_for_online(timeout):
+    logging.info("Trying to connect to tabby")
+
+    health_url = f"http://127.0.0.1:{PORT}/v1/health"
+    
+    is_online = False
+    till = time.time() + timeout * 1000
+
+    while time.time() < till:
+        try:
+            r = requests.post(health_url)
+            if r.status_code == 200:
+                logging.info("Tabby is online now")
+                is_online = True
+                break
+        except:
+            logging.info("Retrying to connect")
+        time.sleep(1)
+    
+    return is_online
+
+
+def index(args):
+    binary = args["tabby_path"]
+    index_repo_url = args["index_repo_url"]
+
+    # Write to config.toml
+    config_file_path = os.path.expanduser("~/.tabby/config.toml")
+    config = {
+        "repositories": [
+            {
+                "git_url": index_repo_url,
+            }
+        ],
+        "experimental": {
+            "enable_prompt_rewrite": True,
+        }
+    }
+    with open(config_file_path, "w+") as f:
+        toml.dump(config, f)
+
+    # Start indexing
+    cmd = [binary, "scheduler", "--now"]
+    subprocess.run(cmd)
+
+def generate_completion_segments(args):
+    sample_repo_url = args["sample_repo_url"]
+    language = args["language"]
+    prompt_count = args["prompt_count"]
+
+    segments = []
+
+    # Checkout the sample repo
+    repo_name = sample_repo_url.split("/")[-1]
+    repo_path = os.path.expanduser(repo_name)
+    if not os.path.exists(repo_path):
+        logging.info("Fetching sample repo")
+        Repo.clone_from(sample_repo_url, repo_path)
+    
+    # Index the files
+    sample_file_name = f"tabby_{language}.index"
+    sample_file_path = os.path.join(repo_path, sample_file_name)
+    if not os.path.exists(sample_file_path):
+        logging.info("Traversing the repo to make sample index")
+        total_lines = 0
+        with open(sample_file_path, "a+") as sample_file_w:
+            for root, _, files in os.walk(repo_path):
```

Comment:
instead of walk the repo, would it be easier to simply read the `dataset/data.jsonl`? You can easily do filtering based on `language` field, instead of maintiaining a separate `LANGUAGE_SUFFIX_MAP`